pkg/networking/trafficplugin/registry/registry.go

```diff
@@ -130,10 +129,24 @@ func (esd *EdgeServiceDiscovery) FindMicroServiceInstances(consumerID, microServ
 	} else {
 		// set Pod ip if hostPort != 0
 		for _, p := range pods {
+			if p.Spec.HostNetwork {
+				// host network
+				hostPort = int32(targetPort)
+			} else {
+				// container network
+				for _, container := range p.Spec.Containers {
+					for _, port := range container.Ports {
+						if port.ContainerPort == int32(targetPort) {
+							hostPort = port.HostPort
```

Comment:
Why do we have to reacquire hostPath here?

Reply:
If not do this, all pods' hostports will be the same even though they  were set different hostports. So by reacquiring hostPath, we can acquire accurate hostport of each pod.

Reply:
/lgtm